amplify/backend/function/fetchMotionState/src/utils.js

```diff
@@ -194,41 +194,49 @@ const updateStakerRewardsInDB = async (
     }),
   );
 
-  await graphqlRequest(updateColonyAction, {
-    id: transactionHash,
-    motionData: {
+  await graphqlRequest(updateColonyMotion, {
+    input: {
       ...motionData,
       stakerRewards: updatedStakerRewards,
     },
   });
 };
 
-const updateMotionMessagesInDB = async (
-  transactionHash,
-  motionData,
-  motionMessages,
-  flag,
-) => {
+const updateMotionMessagesInDB = async (motionData, motionMessages, flag) => {
   const { messages, motionStateHistory } = motionData;
-  const updatedMessages = [...messages];
+  const updatedMessages = [];
   const updatedStateHistory = {
     ...motionStateHistory,
     [flag]: true,
   };
 
   motionMessages.forEach((message) => {
-    updatedMessages.push({
-      initiatorAddress: constants.AddressZero,
-      name: message,
-      messageKey: `${transactionHash}_${message}`,
+    const messageKey = `${motionData.id}_${message}`;
+    if (
+      !messages?.items?.find(
+        (motionMessage) => motionMessage.messageKey === messageKey,
+      )
+    ) {
+      updatedMessages.push({
+        initiatorAddress: constants.AddressZero,
+        name: message,
+        messageKey,
+        motionId: motionData.id,
+      });
+    }
+  });
+
+  const messagePromises = updatedMessages.map((message) => {
```

Comment:
You could probably combine this with the loop above if you use `reduce` on motion messages instead for each. Then await all the promises after.

``` js
const updatedMessages = motionMessages.reduce(() => {})
await Promise.all(updatedMessages)

```

Reply:
Sounds good to me!